clang/lib/Parse/ParseOpenACC.cpp

```diff
@@ -208,6 +233,10 @@ OpenACCDirectiveKind ParseOpenACCDirectiveKind(Parser &P) {
   // introspect on the spelling before then.
   if (FirstTok.isNot(tok::identifier)) {
     P.Diag(FirstTok, diag::err_acc_missing_directive);
+
+    if (!FirstTok.isAnnotation())
+      P.ConsumeAnyToken();
```

Comment:
I'm assuming that's for the end of pragma annotation. Shouldn't the test be more specific?

Reply:
Done!